src/TriggerBinding/SqlTriggerMetricsProvider.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Logging;
+using static Microsoft.Azure.WebJobs.Extensions.Sql.SqlTriggerConstants;
+using static Microsoft.Azure.WebJobs.Extensions.Sql.SqlTriggerUtils;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql
+{
+    internal class SqlTriggerMetricsProvider
```

Comment:
Add doc comment explaining what this class is used for (ideally with links to public docs about scaling)